apps/aecore/lib/aecore/keys/peer.ex

```diff
@@ -0,0 +1,108 @@
+defmodule Aecore.Keys.Peer do
+  @moduledoc """
+    Module for working with Peer keys
+  """
+
+  alias Aewallet.Cypher
+  alias Aecore.Keys.Utils
+  alias Aecore.Keys.Worker, as: Keys
+
+  @type pubkey :: binary()
+
+  @type privkey :: binary()
+
+  @type t :: {pubkey(), privkey()}
+
+  @doc """
+  Gets the default dir for storing the wallet
+  """
+  @spec keypair_dir() :: String.t()
+  def keypair_dir do
+    Application.get_env(:aecore, :peer_keys)[:path]
+  end
+
+  def keypair do
+    Keys.get_peer_keypair()
+  end
+
+  @spec create_keypair() :: :ok | {:error, String.t()}
+  def create_keypair do
+    case keypair_dir()
+         |> File.mkdir()
+         |> Utils.has_dir?(keypair_dir()) do
+      :ok ->
+        :ok
+
+      {:error, :empty} ->
+        keypair = generate_keypair()
+
+        case check_keypair(keypair) do
+          true ->
+            keypair
+            |> encrypt_keypair(Utils.password())
+            |> save_keypair(keypair_dir())
+
+          false ->
+            {:error, "#{__MODULE__} Derivation of Peer keypair failed"}
+        end
+
+      err ->
+        err
+    end
+  end
+
+  def generate_keypair do
```

Comment:
this should be done, by resolving https://github.com/aeternity/elixir-wallet/issues/102

Reply:
Well it could be left like this until I resolve the issue in the elixir-wallet repo

Reply:
okay